packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -319,6 +321,7 @@ function validateRenderResult(child, type) {
 
 function resolve(child, context) {
   while (React.isValidElement(child)) {
+    child = (child: React.Element);
```

Comment:
This cast seemed suspicious to me. Because if it "worked" then Flow would've inferred it anyway.

I checked, and it seems like `React.Element` is `any` 😞 Flow can be so confusing. This is not the thing we're supposed to be using, apparently.

Reply:
Yeah, one of my few wishes for flow is that it would be easier to see what it thought something was at any point in time and why.